checks/raw/sbom_test.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2024 OpenSSF Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package raw
+
+import (
+	"context"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+
+	"github.com/ossf/scorecard/v5/checker"
+	"github.com/ossf/scorecard/v5/clients"
+	mockrepo "github.com/ossf/scorecard/v5/clients/mockclients"
+	"github.com/ossf/scorecard/v5/finding"
+	scut "github.com/ossf/scorecard/v5/utests"
+)
+
+func TestSbom(t *testing.T) {
+	tests := []struct {
+		name     string
+		releases []clients.Release
+		files    []string
+		err      error
+		expected checker.SBOMData
+	}{
+		{
+			name: "With Sbom in release artifacts",
+			releases: []clients.Release{
+				{
+					Assets: []clients.ReleaseAsset{
+						{
+							Name: "test-sbom.cdx.json",
+							URL:  "https://this.url",
+						},
+					},
+				},
+			},
+			files: []string{},
+			expected: checker.SBOMData{
+				SBOMFiles: []checker.SBOM{
+					{
+						Name: "test-sbom.cdx.json",
+						File: checker.File{
+							Type: finding.FileTypeURL,
+						},
+					},
+				},
+			},
+			err: nil,
+		},
+		{
+			name:     "With Sbom in source",
+			releases: []clients.Release{},
+			files:    []string{"test-sbom.spdx.json"},
+			err:      nil,
+			expected: checker.SBOMData{
+				SBOMFiles: []checker.SBOM{
+					{
+						Name: "test-sbom.spdx.json",
+						File: checker.File{
+							Type: finding.FileTypeSource,
+						},
+					},
+				},
+			},
+		},
+		{
+			name:     "Without SBOM",
+			releases: []clients.Release{},
+			files:    []string{},
+			expected: checker.SBOMData{},
+			err:      nil,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt // Re-initializing variable so it is not changed while executing the closure below
+		t.Run(tt.name, func(t *testing.T) {
+			t.Setenv("SCORECARD_EXPERIMENTAL", "true")
```

Comment:
the env var isn't needed for the raw function and can be removed.

(the linter will flag you for `t.Parallel` once you remove it, so fix that)